src/base/profile.cpp

```diff
@@ -33,29 +33,27 @@
 
 Profile *Profile::m_instance = nullptr;
 
-Profile::Profile(Private::Profile *impl, Private::PathConverter *pathConverter)
-    : m_profileImpl(impl)
-    , m_pathConverterImpl(pathConverter)
+Profile::Profile(const QString &rootProfilePath, const QString &configurationName, bool convertPathsToProfileRelative)
 {
+    m_profileImpl = rootProfilePath.isEmpty()
+                    ? static_cast<Private::Profile *>(new Private::DefaultProfile(configurationName))
+                    : static_cast<Private::Profile *>(new Private::CustomProfile(rootProfilePath, configurationName));
+
     ensureDirectoryExists(SpecialFolder::Cache);
     ensureDirectoryExists(SpecialFolder::Config);
     ensureDirectoryExists(SpecialFolder::Data);
+
+    m_pathConverterImpl = convertPathsToProfileRelative
+                          ? static_cast<Private::PathConverter *>(new Private::Converter(m_profileImpl->baseDirectory()))
+                          : static_cast<Private::PathConverter *>(new Private::NoConvertConverter());
 }
 
 // to generate correct call to ProfilePrivate::~ProfileImpl()
 Profile::~Profile() = default;
 
-void Profile::initialize(const QString &rootProfilePath, const QString &configurationName,
-                         bool convertPathsToProfileRelative)
+void Profile::initialize(const QString &rootProfilePath, const QString &configurationName, bool convertPathsToProfileRelative)
```

Comment:
`const bool convertPathsToProfileRelative` (only in here, not in header)

Reply:
Thanks for the fixes, pushed these fixes already.